hw/ip/aes/pre_sca/alma/cpp/testbench.h

```diff
@@ -0,0 +1,73 @@
+// Copyright IAIK.
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "verilated.h"
+#include "verilated_vcd_c.h"
+
+template <class Module>
+struct Testbench {
+  unsigned long m_tickcount;
+  Module *m_core;
+  VerilatedVcdC *m_trace;
+
+  Testbench(void) {
+    Verilated::traceEverOn(true);
+    m_core = new Module;
+    m_tickcount = 0ul;
+  }
+
+  virtual ~Testbench(void) {
+    delete m_core;
+    m_core = NULL;
+  }
+
+  virtual void opentrace(const char *vcdname) {
```

Comment:
Any particular reason to make these all virtual? If not, I'd probably drop the `virtual` keywords because it makes the code a bit easier to reason about.

Reply:
Nope, I've taken this code from an example in the Alma repo. Now removed the use of `virtual`.